nmigen/build/run.py

```diff
@@ -149,8 +149,10 @@ def mkdirs(path):
                 sftp.chdir(root)
                 for filename, content in self.files.items():
                     filename = pathlib.PurePosixPath(filename)
-                    # Just to make sure we don't accidentally overwrite anything outside of build root.
-                    assert not filename.parts[0] == ".."
+                    # Forbid parent directory components completely to avoid the possibility
+                    # of writing outside the build root. We can't use normpath to remove ".."
+                    # because we may be running on Windows, but the remote is a *nix machine.
+                    assert ".." not in filename.parts
```

Comment:
Instead of explaining why we don't use normpath here, let's just drop it in local builds as well. This way we don't have to explain anything, and the next person who copies the code for local builds doesn't introduce a similar bug, either.

Reply:
Fixed. `os.path` functions accepting `Pathlib` objects works down to and include Python 3.6, which is our cutoff. So might as well use it.